plotutil/xyer.go

```diff
@@ -0,0 +1,77 @@
+package plotutil
+
+import "github.com/gonum/plot/plotter"
+
+type xyer struct {
+	x, y []float64
+	len  int
+}
+
+// Len returns the length of the XYer object.
+// fulfills the plotter.XYer interface requirement.
+func (xy *xyer) Len() int {
+	return xy.len
+}
+
+// XY returns the x,y values at index idx.
+// fulfills the plotter.XYer interface requirement.
+func (xy *xyer) XY(idx int) (x, y float64) {
+	return xy.x[idx], xy.y[idx]
+}
+
+// ZipXY is a convenience function to build a plotter.XYer from slices.
+// Length will be set to the shortest slice length.
+// Panics on empty or nil slices.
+func ZipXY(x, y []float64) plotter.XYer {
```

Comment:
The implementation decisions were focused as much as possible on the "convenience" aspect.  It's not meant to be a replacement data structure for `plotter.XYs` or anything like that.  Each object is only expected to exist long enough to plot the target data.  That's why the return type is the interface itself, users should make no assumptions about the underlying implementation.  The struct type isn't exported, so there's no way to type assert from the interface without unsafe/reflect hacks.  That's also why I used the 'Zip' name, so users will expect the result will be zipped touples (x[i], y[i], <z[i]>) from slice parameters.  There's no need and limited ability to dig into the guts of the resulting object.  If that's what they need, a call to `plotter.CopyXYs(plotutil.ZipXY(x,y))` does the job with the objects that already exist.

The target is where a `plotter.XYer` or `plotter.XYZer` interface type is required and the data isn't necessarily in a data structure that easily implements the interface for each combination of slices.  

Something like:
```go
var data *mat64.Dense = loadData("file.csv")
times := data.RawRowView(0)
rowlen,_ := data.Dims()

for i:=1; i < rowlen; i++ {
        line, _ := plotter.NewLine(plotutil.ZipXY(times, data.RawRowView(i)))
        line.Plot(canvas, plot)
}
```

I used the shortest length in order to allow for ragged matrix structures without needing to juggle lengths.  I was trying to make it as convenient/flexible as possible, while only `panic`-ing on empty (0-length) returns that wouldn't show up on the resulting plot.